notebooks/Evaluation - noise.ipynb

```diff
@@ -0,0 +1,2261 @@
+      "{'dataset': 'eurosat_rgb', 'net': 'efficientnet-b0', 'batch': 32, 'confidence': 0.95, 'lr': 0.03, 'uratio': 7, 'wd': 0.00075, 'wu': 1.0, 'seed': 4, 'numlabels': 100, 'opt': 'SGD', 'iterations': 0, 'batch_size': 256, 'data_dir': './data/', 'use_train_model': False, 'load_path': '/scratch/fixmatch_results/new_runs/nr_of_labels/eurosat_rgb/FixMatch_archefficientnet-b0_batch32_confidence0.95_lr0.03_uratio7_wd0.00075_wu1.0_seed4_numlabels100_optSGD/model_best.pth'}\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "results = []\n",
+    "for checkpoint, args in zip(checkpoints,run_args):\n",
+    "    print(\"------------ RUNNING \", checkpoint, \" -----------------\")\n",
+    "    print(args)\n",
+    "    args[\"batch_size\"] = 256\n",
+    "    args[\"data_dir\"] = \"./data/\"\n",
+    "    args[\"use_train_model\"] = False\n",
+    "    args[\"load_path\"] = checkpoint\n",
+    "    \n",
+    "    if args[\"seed\"] == seed_wanted:\n",
+    "        checkpoint_path = os.path.join(args[\"load_path\"])\n",
+    "        checkpoint = torch.load(checkpoint_path,map_location='cuda:0')\n",
+    "        load_model = (checkpoint[\"train_model\"] if args[\"use_train_model\"] else checkpoint[\"eval_model\"])\n",
+    "        _net_builder = net_builder(args[\"net\"],False,{})\n",
+    "        _eval_dset = SSL_Dataset(name=args[\"dataset\"], train=False, data_dir=args[\"data_dir\"], seed=args[\"seed\"])\n",
+    "        eval_dset = _eval_dset.get_dset()\n",
+    "        inv_transf = _eval_dset.inv_transform\n",
+    "        transf = _eval_dset.transform\n",
+    "        net = _net_builder(num_classes=_eval_dset.num_classes, in_channels=_eval_dset.num_channels)\n",
+    "        net.load_state_dict(load_model)\n",
+    "        if torch.cuda.is_available():\n",
+    "            net.cuda()\n",
+    "        net.eval()\n",
+    "                        \n",
+    "        eval_loader = get_data_loader(eval_dset, args[\"batch\"], num_workers=1)\n",
+    "        label_encoding = _eval_dset.label_encoding\n",
+    "        dB_value_list = []\n",
+    "        resuts_run = []\n",
+    "        \n",
+    "        for P_noise_dB in range(P_noise_dB_min, P_noise_dB_max, dP_noise_dB):\n",
+    "            print(\"------------ PREDICTING TESTSET ----------------- - Noise value [dB]\", P_noise_dB)\n",
+    "            \n",
+    "            \n",
+    "            images, labels, preds = [],[],[]\n",
+    "            with torch.no_grad():\n",
+    "                for image, target in tqdm(eval_loader):\n",
+    "                    image_preprocessed=[]\n",
+    "                    for idx,img in enumerate(image):\n",
+    "                        img=255 * inv_transf(img.transpose(0,2).cpu().numpy())\n",
+    "                        img_orig=img\n",
+    "                        for n in range(img.shape[0]):\n",
```

Comment:
I don't think this is the channel? two lines above with the transpose the image is changed from (channel,width,heigth) to (width,height,channel) So you are scaling each column differently I think?

Reply:
It is, I printed the img shape just after the for:

![image](https://user-images.githubusercontent.com/70584239/134551944-8ea5f948-79f4-49d6-9652-2c24b5f1cfa7.png)